pyquil/paulis.py

```diff
@@ -22,17 +22,20 @@
 import numpy as np
 import copy
 
-from typing import Dict, FrozenSet, Iterable, Iterator, List, Optional, Tuple, Union
+from typing import Callable, Dict, FrozenSet, Iterable, Iterator, List, Optional, Sequence, Tuple, Union
+from pyquil.quilatom import ExpressionValueDesignator
 
 from pyquil.quilatom import QubitPlaceholder
 
 from .quil import Program
 from .gates import H, RZ, RX, CNOT, X, PHASE, QUANTUM_GATES
 from numbers import Number
 from collections import OrderedDict
-from collections.abc import Sequence
 import warnings
 
+PauliDesignator = Union['PauliTerm', 'PauliSum']
+PauliT = Union[PauliDesignator, ExpressionValueDesignator]
```

Comment:
let's remove `PauliT` and replace all of its uses with `Union[PauliDesignator, PauliCoefficientDesignator]`